yktrace/src/hwt/mod.rs

```diff
@@ -38,6 +38,14 @@ impl ThreadTracerImpl for HWTThreadTracer {
     }
 }
 
+impl Drop for HWTThreadTracer {
+    fn drop(&mut self) {
+        // If we haven't stopped the tracer yet, do it now. This might return an error if the
+        // tracer was already stopped, but we don't care as long as it's stopped.
```

Comment:
This is not necessary for safety, right? `Drop` is not guaranteed to be called. `mem::forget` is safe.

Reply:
It's to do with a bug, where the tracer was dropped while it was still tracing, which left the tracer running, and meant other threads weren't allowed to trace. I'm not sure we can do anything about users `mem:forget`ting the tracer, but if they do that, then they shouldn't be surprised if stuff breaks.

Reply:
`mem::forget` causing logic bugs is fine, but it must not lead to memory unsafety.

Reply:
If `drop` isn't called, PT will keep on running pointlessly but harmlessly (other than consuming memory + power), so we should be fine here I believe.